src/main/java/org/codehaus/plexus/build/DefaultBuildContext.java

```diff
@@ -134,28 +153,35 @@ public void addWarning(File file, int line, int column, String message, Throwabl
         addMessage(file, line, column, message, SEVERITY_WARNING, cause);
     }
 
+    private String getMessage(File file, int line, int column, String message) {
+        return file.getAbsolutePath() + " [" + line + ':' + column + "]: " + message;
+    }
+
     /** {@inheritDoc} */
     public void addMessage(File file, int line, int column, String message, int severity, Throwable cause) {
-        switch (severity) {
-            case BuildContext.SEVERITY_ERROR:
-                logger.error(getMessage(file, line, column, message), cause);
-                return;
-            case BuildContext.SEVERITY_WARNING:
-                logger.warn(getMessage(file, line, column, message), cause);
-                return;
+        if (isDefaultImplementation()) {
+            switch (severity) {
+                case BuildContext.SEVERITY_ERROR:
+                    logger.error(getMessage(file, line, column, message), cause);
+                    return;
+                case BuildContext.SEVERITY_WARNING:
+                    logger.warn(getMessage(file, line, column, message), cause);
+                    return;
+                default:
+                    logger.debug(getMessage(file, line, column, message), cause);
+                    return;
+            }
         }
-        throw new IllegalArgumentException("severity=" + severity);
+        legacy.addMessage(file, line, column, message, severity, cause);
     }
 
     /** {@inheritDoc} */
-    public void removeMessages(File file) {}
+    public void removeMessages(File file) {
+        legacy.removeMessages(file);
```

Comment:
Just we have `if (!isDefaultImpementation())` just to be consistent with the `addMessage` ones ?